File: APCS202301_造字程式-2.cpp
#include <bits/stdc++.h>

using namespace std;

#define N 10005
typedef long long ll;

int main()
{
    // freopen("P_7_12_2.in", "r", stdin);
    int k, q, r;
    cin >> k >> q >> r;
    string s;
    cin >> s;
    s = '0' + s;
    vector<string> ret;
    for (int i = 0; i < q; ++i)
    {
        string t(k + 1, '0');
        int newidx;
        for (int idx = 1; idx <= k; ++idx)
        {
            cin >> newidx;
            t[newidx] = s[idx];
        }
        ret.push_back(t);
        swap(s, t);
    }
    for (int i = 1; i <= r; ++i)
    {
        for (string e : ret)
            cout << e[i];
        cout << '\n';
    }
    return 0;
}

File: APCS202301_造字程式-2.py
k, q, r = map(int, input().split())
s = '0'+input()
ret = []
for _ in range(q):
    p = [0]+[int(x) for x in input().split()]
    tmp = ['0']*(k+1)
    for idx, newidx in enumerate(p):
        tmp[newidx] = s[idx]
    s = "".join(tmp)
    ret.append(s)
ret = list(zip(*ret))
for i in range(1, r+1):
    print("".join(ret[i]))
